src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int sumSupply = 0;
+        int sumBuy = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String [] dataLine = line.split(",");
+                if (dataLine[0].equals("supply")) {
+                    sumSupply += Integer.parseInt(dataLine[1]);
+                }
+                if (dataLine[0].equals("buy")) {
+                    sumBuy += Integer.parseInt(dataLine[1]);
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Error reading file: " + e.getMessage());
```

The current implementation violates the 'Don't ignore exceptions' rule. Instead of just printing the error message, re-throw `RuntimeException` with the original exception and a descriptive message, like this:

```java
throw new RuntimeException("Can't read data from the file " + fromFileName, e);
```